internal/controllers/gateway/route_utils.go

```diff
@@ -630,3 +630,12 @@ func isHTTPReferenceGranted(grantSpec gatewayv1alpha2.ReferenceGrantSpec, backen
 	}
 	return false
 }
+
+// sameCondition returns true if the condition has the same type, status, reason and message
+// with expectedCondition in conditions.
+func sameCondition(expectedCondition metav1.Condition, condition metav1.Condition) bool {
```

Comment:
It'd idiomatic in Go to just call the params `a` and `b` or something else but at this level of conciseness

```suggestion
func sameCondition(a, b metav1.Condition) bool {
```

Reply:
updated.